xep-0425.xml

```diff
@@ -84,30 +97,29 @@
 </message>]]></example>
 
   <p>The moderator sees the message and indicates to their client that the message should be retracted. The client then sends an IQ stanza to the MUC service, requesting that the message be retracted.</p>
+  <p>An optional &lt;reason/&gt; element may be included inside the &lt;moderate/&gt; element, to provide a human-readable explanation why the message was retracted.</p>
 
   <example caption="The moderator's client sends an IQ stanza to the MUC service"><![CDATA[
 <iq type='set' to='room@muc.example.com' id='retract-request-1'>
-  <apply-to id="stanza-id-1" xmlns="urn:xmpp:fasten:0">
-    <moderate xmlns='urn:xmpp:message-moderate:0'>
-      <retract xmlns='urn:xmpp:message-retract:0'/>
-      <reason>This message contains inappropriate content for this forum</reason>
-    </moderate>
-  </apply-to>
+  <moderate id="stanza-id-1" xmlns='urn:xmpp:message-moderate:1'>
+    <retract xmlns='urn:xmpp:message-retract:1'/>
+    <reason>This message contains inappropriate content for this forum</reason>
+  </moderate>
 </iq>]]></example>
 
   <section2 topic='Success case' anchor='usecase-success'>
 
   <p>If the moderator is allowed to moderate the message, the groupchat service will send a message stanza to all participants (including the moderator), indicating that the message has been retracted and by whom.</p>
-  <p>This message will use &xep0422; to indicate that it applies to the moderated message, by referring to its XEP-0359 Stanza ID.</p>
+  <p>This message will use &xep0424; to indicate that it is a retraction, and will refer to the retracted message by its XEP-0359 Stanza ID. Care must be taken to use the stanza ID that was added by the MUC, i.e. which has a "by" attribute set to the MUC JID, and not some other stanza ID. The client MUST check that the correct stanza ID was used and ignore the stanza if not.</p>
+  <p>Inside the &lt;retract/&gt; element is a &lt;moderated/&gt; element indicating that the retraction was a moderator action and containing the optional reason.</p>
```

Comment:
Please describe the elements that should or must be in the message here when the message is first introduced

- Is the "by" attributed a MUST, or optional?
- Is the "by" attribute set to a real JID in non-anon rooms? Or always the occupant JID?
- Should a server that understands the occupant-id XEP add a <occupant-id> tag inside the moderated tag?

Some of these things are defined later down in the Tombstone section, which someone may not even read because they have no interest in implementing tombstones, and my interpret only relevant for the tombstone use case.